builtins: implement ST_ConvexHull

otan opened the pull request:
Resolves #48909.

Release note (sql change): Implement the ST_ConvexHull function on
geometry types.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/51446)
<!-- Reviewable:end -->

otan committed:
builtins: implement ST_ConvexHull

Resolves #48909.

Release note (sql change): Implement the ST_ConvexHull function on
geometry types.

rytaft reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 8 of 8 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/51446#-:-MCICrdp8jjdARL0Zibm:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

otan commented:
TFTR!
bors r=rytaft
